src/Data/Void.purs

```diff
@@ -2,11 +2,23 @@ module Data.Void (Void, absurd) where
 
 import Data.Show (class Show)
 
--- | An uninhabited data type.
+-- | An uninhabited data type. In other words, one can never create
+-- | a runtime value of type `Void` becaue no such value exists.
 -- |
 -- | `Void` is useful to eliminate the possibility of a value being created.
 -- | For example, a value of type `Either Void Boolean` can never have
 -- | a Left value created in PureScript.
+-- |
+-- | This should not be confused with the word, `void,` that commonly appears in
+-- | C-deriving languages, such as Java:
+-- | ```
+-- | public class Foo {
+-- |   void doSomething() { System.out.println("hello world!"); }
+-- | }
+-- | ```
+-- |
+-- | In PureScript, one often uses `Unit` to achieve similar effects as
+-- | the lowercased `void` above.
```

Comment:
I think it would be slightly clearer to say "the `void` of C-family languages" here, so that the reader doesn't have to double-check the casing to see which `void` is which in order to know which one is meant here.

Reply:
Good point. Done